### PR TITLE
Round: 10: performance improvements

### PR DESCRIPTION
**Pull request by jasnell**
The start of round 10....

### CODE REVIEW

File: src/node-http2-core-inl.h [hunk 1]
```diff
@@ -195,6 +194,13 @@ inline bool nghttp2_session_find_stream(
   }
 }
 
+inline void nghttp2_set_callbacks_allocate_send_buf(
+    node_nghttp2_session_callbacks* callbacks,
+    nghttp2_allocate_send_buf_cb cb) {
+  assert(callbacks != nullptr);
+  callbacks->allocate_send_buf = cb;
+}
```
Comment (addaleax) [inline src/node-http2-core-inl.h:202]: I find these functions a bit curious… what are the reasons not to inline them at their call sites, or at least make them methods on `node_nghttp2_session_callbacks`?

File: src/stream_base.h [hunk 2]
```diff
@@ -93,6 +93,17 @@ class WriteWrap: public ReqWrap<uv_write_t>,
```
Comment (addaleax) [inline src/stream_base.h:93]: What’s up with these file access mode changes? :D

Comment (jasnell) [reply src/stream_base.h:93]: Windows and I keep forgetting to tell git to ignore them.

File: src/node-http2-core-inl.h (hunk 1 above)
Comment (jasnell) [reply src/node-http2-core-inl.h:202]: started this by following the pattern nghttp uses but I've been wanting to refactor it to simplify.

**Comment by jasnell** [comment]
Landed

**Comment by mcollina** [comment]
🎉

**Comment by jasnell** [comment]
Will start on round 11 next week as time allows. Also @addaleax has offered to do some digging into the code also... I'm quite certain she'll be able to find many ways of improving it further :-)

**Pull request closed without merging**